### DISCUSSION TITLE
Creating custom error messages

### DISCUSSION THREAD

**Discussion by Inayat-C** [prisma/prisma#6183] [Q&A]
Hi,

Let's say we are creating a new database record for a user where the email address field has a unique constraint.

```ts
const newUser = await prisma.user.create({
    data: {
        email,
        password: passwordHash,
    },
});
```

If the e-mail address isn't unique, we get an error:

```
{
    "code": "P2002",
    "clientVersion": "2.18.0",
    "meta": {
        "target": [
            "email"
        ]
    }
}
```

What if I want to create my own custom error based on the error handling API I use, something like so:

```ts
await prisma.user.create({
    data: {
        email,
        password: passwordHash,
    },
}).then(() => {
    next();
})
.catch((e) => {
    next(new BadRequest(e.msg));
});
```

Is something like this possible to do? I know the error object above doesn't have a message property but on the console when I log the error I can see this message:

```
Invalid `prisma.user.create()` invocation:
Unique constraint failed on the fields: (`email`)
```

Is there anyway to get a hold of that message on the error object?

**Comment by ryands17** [comment]
@Inayat-C 👋 

Currently you can get the [error code and message](https://www.prisma.io/docs/reference/api-reference/error-reference) using the `e.code` and `e.meta ` property. This will get you the message and the field that was violated by the unique constraint.

Let me know if that's what you were looking for :)

**Reply by Inayat-C** [reply]
I wasn't able to get the actual error message from `e.code` or `e.meta`. But I managed to solve my problem this way:

```ts
const emailErrorMessage = 'An account using this email address has already been registered';

try {
    // Creates a new user record in the database
    const newUser = await prisma.user.create({
        data: {
            email
        },
    });

    // Sends email and user ID back to client
    res.status(200).json({
        email: newUser.email,
        userID: newUser.id,
    });
} catch (err) {
    err.code === 'P2002' ? nxt(new BadRequest(emailErrorMessage)) : nxt(err);
}
```

An issue here is what if a different Prisma error code is returned?

**Reply by ryands17** [reply]
The actual error message will be obtained from `e.message`.

> An issue here is what if a different Prisma error code is returned?

This error code will always be the same for a unique constraint error.

**Reply by Inayat-C** [reply]
Ah, that was what I was looking for! Wasn't clear in the docs that I can access the `e.message` property.

Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
